backend/src/main/java/dccs/academy/services/SupplierService.java

```diff
@@ -0,0 +1,36 @@
+package dccs.academy.services;
+
+import dccs.academy.dtos.SupplierDto;
+import dccs.academy.repositories.SupplierRepository;
+import dccs.academy.utils.mappers.SupplierMapper;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+@Transactional
+public class SupplierService {
+    @Inject
+    SupplierRepository supplierRepository;
+    public void createSupplier(SupplierDto supplierDto) {
+        var supplierEntity = SupplierMapper.toEntity(supplierDto);
+        supplierRepository.persist(supplierEntity);
+    }
+
+    public List<SupplierDto> supplierSearch(String names, String city, Long id){
+        String query = "lower(names) like ?1 and lower(city) like ?2";
```

Comment:
Criteria API please instead of these queries because you here still have hardcoded field names and this is bad practice.

Also Criteria API query should be written in repo class.